workflows/taxprofiler.nf

```diff
@@ -20,6 +22,13 @@ if (params.databases) { ch_databases = file(params.databases) } else { exit 1, '
 if (params.shortread_clipmerge_mergepairs && params.run_malt ) log.warn "[nf-core/taxprofiler] warning: MALT does not accept uncollapsed paired-reads. Pairs will be profiled as separate files."
 if (params.shortread_clipmerge_excludeunmerged && !params.shortread_clipmerge_mergepairs) exit 1, "[nf-core/taxprofiler] error: cannot include unmerged reads when merging not turned on. Please specify --shortread_clipmerge_mergepairs"
 
+// TODO Add check if index but no reference exit 1
+if (params.shortread_hostremoval && !params.shortread_hostremoval_reference) { exit 1, "[nf-core/taxprofiler] error: --shortread_hostremoval requested but no --shortread_hostremoval_reference FASTA supplied. Check input." }
+if (!params.shortread_hostremoval_reference && params.shortread_hostremoval_reference_index) { exit 1, "[nf-core/taxprofiler] error: --shortread_hostremoval_index provided but no --shortread_hostremoval_reference FASTA supplied. Check input." }
+
+if (params.shortread_hostremoval_reference ) { ch_reference       = file(params.shortread_hostremoval_reference) } else { ch_reference       = [] }
+if (params.shortread_hostremoval_index     ) { ch_reference_index = file(params.shortread_hostremoval_index    ) } else { ch_reference_index = [] }
```

Comment:
It is not exactly a review but rather a question of what is the idea behind these two lines :)

Reply:
That.... is a good question 😂

At least L30 makes sense - we need both the reference and index channels to execute the hostremoval sub-workflow. 
However, if the user wants the pipeline to generate the reference index, they would not supply the index. If we didn't have L30, the hostremoval subworkflow would not be executed because not both input channels would be provided.

Therefore in this case, we  just put an empty list if the index is not supplied. Therefore the input channels are both satisfied, but as it's 'empty' for index we will generate that for the user.

I'll check if L29 is still necessary :+1:

Reply:
Ok L29 was partially necessary, had to at least make the input channel for the reference (if supplied), however we didn't need the `else` statement. However the refernece index channel IS optional, so we have to make an empty channel if the user doesn't pass it to hte pipeline.

Reply:
Sounds good to me.